app/services/billable_metrics/aggregation_factory.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module BillableMetrics
+  class AggregationFactory
+    def self.new_instance(charge:, current_usage: false, **attributes)
+      aggregator_class(charge, current_usage).new(
+        billable_metric: charge.billable_metric,
+        **attributes
+      )
+    end
+
+    private
+
+    def self.aggregator_class(charge, current_usage)
```

Comment:
Lint/IneffectiveAccessModifier: `private` (on line 12) does not make singleton methods private. Use `private_class_method` or `private` inside a `class << self` block instead.